luxtronik/discover.py

```diff
@@ -0,0 +1,56 @@
+"""Discovery funtion for Luxtronik2 heatpump controllers."""
+import logging
+import socket
+
+LOGGER = logging.getLogger("Luxtronik.Discover")
+
+
+def discover() -> list[(str, int)]:
+    """Broadcast discovery for luxtronik heatpumps."""
+
+    results: list[(str, int)] = list()
+
+    for magic_port in (4444, 47808):
+        LOGGER.debug("Send discovery packets to port %s", magic_port)
+        server = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
+        server.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+        server.bind(("", magic_port))
+        server.settimeout(2)
+
+        # send AIT magic broadcast packet
+        data = "2000;111;1;\x00"
+        server.sendto(data.encode(), ("<broadcast>", magic_port))
+        LOGGER.debug("Sending broadcast request %s", data.encode())
+
+        while True:
+            try:
+                res, con = server.recvfrom(1024)
+                res = res.decode("ascii", errors="ignore")
+                # if we receive what we just sent, continue
+                if res == data:
+                    continue
+                ip_address = con[0]
+                # if the response starts with the magic nonsense
+                if res.startswith("2500;111;"):
+                    res = res.split(";")
+                    LOGGER.debug("Received answer from %s %s", ip_address, str(res))
+                    try:
+                        port = int(res[2])
```

Comment:
Do we need any kind of sanity check on this value? Maybe we should check whether the int value is in between `1-65535`?